ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -715,8 +715,13 @@ regionSum(const ScalarBuffer& property,
         if (property.empty())
             return totals;
 
-        assert(regionId.size() == property.size());
-        for (std::size_t j = 0; j < regionId.size(); ++j) {
+        // the regionId contains the ghost cells
+        // the property does not contain the ghostcells
+        // This code assumes that that the ghostcells are
+        // added after the interior cells
+        // Is this a valid assumption?
```

Comment:
The assumption (interior cells stored before ghost cells) holds unless the user changes the `OwnerCellsFirst` runtime parameter from its default state of `true`.

Reply:
I think testing and throwing if not the case may be in order then? Or even removing the OwnerCellsFirst runtime option.

Reply:
Just remove the option.

Reply:
I correct my statement: Maybe someone should first test what happens with grids apart from CpGrid.

Reply:
The linear solver has
```
            const bool ownersFirst = EWOMS_GET_PARAM(TypeTag, bool, OwnerCellsFirst);
            if (!ownersFirst) {
                const std::string msg = "The linear solver no longer supports --owner-cells-first=false.";
                if (on_io_rank) {
                    OpmLog::error(msg);
                }
                OPM_THROW_NOLOG(std::runtime_error, msg);
            }
```
I can add something similar to the output code.